### PR TITLE
'class' as a object property name

### PR DESCRIPTION
**Pull request by mhnaeem**
Checking to make sure that the token with text 'class' or 'interface' has whitespace after it to be considered a block statement, otherwise treat as object literal

# Description
- [X] Source branch in your fork has meaningful name (not `main`)

Fixes Issue: 
#1838

### CODE REVIEW

File: test/data/javascript/tests.js [hunk 1]
```diff
@@ -230,7 +230,18 @@ exports.test_data = {
...
+      tests: [{
+          comment: "#1838 - handle class word as an object property",
+          unchanged: [
+            '{\n',
+            '    class: {\n',
+            '        a: 1,\n',
+            '        b: 2,\n',
+            '        c: 3,\n',
+            '    }\n',
+            '}'
```
Comment (bitwiseman) [inline test/data/javascript/tests.js:242]: Please move your test input to the regression test area down around line 3500 or so.  This doesn't relate to "Object literal shorthand functions".

File: js/src/javascript/beautifier.js [hunk 2]
```diff
@@ -691,10 +691,10 @@ Beautifier.prototype.handle_start_block = function(current_token) {
     )) {
     // We don't support TypeScript,but we didn't break it for a very long time.
     // We'll try to keep not breaking it.
-    if (!in_array(this._last_last_text, ['class', 'interface'])) {
-      this.set_mode(MODE.ObjectLiteral);
-    } else {
+    if (in_array(this._last_last_text, ['class', 'interface']) && !!second_token.whitespace_before) {
```
Comment (bitwiseman) [inline js/src/javascript/beautifier.js:694]: I don't think whitespace is a good indicator for what you're trying to do here. 

The following two inputs should produce the same output, but I don't think they do with this change:
* `var a={class:{a:1}}`
* `var a = { class : { a : 1 } }`

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Thanks for submitting this PR!
It is a good quality PR, but still needs some changes.

File: js/src/javascript/beautifier.js (hunk 2 above)
Comment (mhnaeem) [reply js/src/javascript/beautifier.js]: @bitwiseman Both the cases you have stated resulted in the same output using whitespace however I took your comments into consideration and changed the condition to look for the text for the `second_token`.

File: test/data/javascript/tests.js (hunk 1 above)
Comment (mhnaeem) [reply test/data/javascript/tests.js]: I wasn't sure about this, thanks for letting me know. I have moved them under regression tests and also added more tests.

File: test/data/javascript/tests.js [hunk 3]
```diff
@@ -3304,6 +3293,42 @@ exports.test_data = {
...
+        },
+        {
+          comment: "#1838 - handle class word as an object property but with space after colon",
+          input: [
+            '{',
+            '    class : { a: 1,',
+            'b: 2,c : 3',
+            '    }',
+            '}'
+          ],
```
Comment (bitwiseman) [inline test/data/javascript/tests.js:3321]: This is good.  I think you need to test at least two more inputs:
* `{class:{a:1,b:2,c:3,}}`
* `{x:{a:1,class:2,c:3,}}`

**Review by bitwiseman — CHANGES_REQUESTED** [review]
A couple more tests and we can merge this.

**Review by bitwiseman — APPROVED** [review]

**Comment by bitwiseman** [comment]
Great work! Thanks!

**Pull request merged**